pkg/ext/delegate_error_test.go

```diff
@@ -0,0 +1,1109 @@
+					Reason: metav1.StatusReasonNotFound,
+				},
+			},
+		},
+		{
+			name:  "generic error",
+			input: assert.AnError,
+			output: &apierrors.StatusError{ErrStatus: metav1.Status{
+				Status: metav1.StatusFailure,
+				Code:   http.StatusInternalServerError,
+				Reason: metav1.StatusReasonInternalError,
+				Details: &metav1.StatusDetails{
+					Causes: []metav1.StatusCause{{Message: assert.AnError.Error()}},
+				},
+				Message: fmt.Sprintf("Internal error occurred: %v", assert.AnError),
+			}},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output := convertError(tt.input)
+			assert.Equal(t, tt.output, output)
+		})
+	}
+
+}
+
+func TestDelegateError_Watch(t *testing.T) {
+	type input struct {
+		ctx             context.Context
+		internaloptions *metainternalversion.ListOptions
+	}
+
+	type output struct {
+		watch watch.Interface
+		err   error
+	}
+
+	type testCase struct {
+		name                    string
+		input                   input
+		expected                output
+		storeSetup              func(*MockStore[*TestType, *TestTypeList])
+		simulateConversionError bool
+		wantedErr               bool
+	}
+
+	tests := []testCase{
+		{
+			name: "missing user in context",
+			input: input{
+				ctx:             context.TODO(),
+				internaloptions: &metainternalversion.ListOptions{},
+			},
+			wantedErr:  true,
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {},
+		},
+		{
+			name: "convert list error",
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				internaloptions: &metainternalversion.ListOptions{},
+			},
+			simulateConversionError: true,
+			wantedErr:               true,
+			storeSetup:              func(ms *MockStore[*TestType, *TestTypeList]) {},
+		},
+	}
+
+	for _, tt := range tests {
+		scheme := runtime.NewScheme()
+		addToSchemeTest(scheme)
+
+		if !tt.simulateConversionError {
+			scheme.AddConversionFunc(&metainternalversion.ListOptions{}, &metav1.ListOptions{}, convert_internalversion_ListOptions_to_v1_ListOptions)
+		}
+
+		gvk := schema.GroupVersionKind{Group: "ext.cattle.io", Version: "v1", Kind: "TestType"}
+		gvr := schema.GroupVersionResource{Group: "ext.cattle.io", Version: "v1", Resource: "testtypes"}
+
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		mockStore := NewMockStore[*TestType, *TestTypeList](ctrl)
+
+		testDelegate := &delegateError[*TestType, *TestTypeList]{
+			inner: &delegate[*TestType, *TestTypeList]{
+				scheme: scheme,
+				t:      &TestType{},
+				tList:  &TestTypeList{},
+				gvk:    gvk,
+				gvr:    gvr,
+				store:  mockStore,
+			},
+		}
+
+		watch, err := testDelegate.Watch(tt.input.ctx, tt.input.internaloptions)
+		if tt.wantedErr {
+			// check if we have an error
+			assert.Error(t, err)
+
+			// check if error is an api error
+			if _, ok := err.(apierrors.APIStatus); ok {
+				assert.True(t, ok)
+				return
+			}
```

Comment:
Should this be

```suggestion
				_, ok := err.(apierrors.APIStatus)
				assert.True(t, ok)
```

? Right now if the error returned is not an APIStatus there just won't be any assert.

Reply:
You're absolutely right. Just sent fixed for all tests. Thank you!